### ISSUE TITLE
[Feature Request] PHP 8.1 compatibility

### ISSUE THREAD

**Issue by ThomasLandauer** [barbushin/php-imap#624]
What's the plan/status for upgrading to PHP 8.1?
Here's what probably needs to be fixed: https://php.watch/versions/8.1/IMAPConnection-resource

**Comment by Sebi94nbg** [comment]
Ah, I was just so happy, that PHP 8.0 support has been added and now there is again such a "big" project...

![Man Crying](https://i.giphy.com/media/s4GPaO3Biktkk/giphy.webp)

I don't have any status or plans regarding this yet as I wanted to clean up the open pull requests and issues first, which... will take some time.

But as always: Everyone is welcome to create pull requests to support this project. :)

**Comment by Sebi94nbg** [comment]
@ThomasLandauer Looks like as some de-/encoding stuff breaks with PHP 8.1:
```shell
Mailbox (PhpImap\Mailbox)
 ✘ Encoding to utf 7 decode back to utf 8 with data set "chiCheŵa" [0.20 ms]
   │
   │ Failed asserting that two strings are equal.
   │ --- Expected
   │ +++ Actual
   │ @@ @@
   │ -'chiCheÅµa'
   │ +'chiCheŵa'
   │
   │ /home/runner/work/php-imap/php-imap/tests/unit/MailboxTest.php:443
   │
```
See https://github.com/barbushin/php-imap/runs/4415332297.

**Comment by ThomasLandauer** [comment]
Sorry, don't know what's going on there :-(

**Comment by nunoperalta** [comment]
@Sebi94nbg @MaSpeng Any estimated date when the new version/package will be released? :)
Thank you very much! Good day.

**Comment by MaSpeng** [comment]
@nunoperalta I am sorry but I only provided the pr for the compatibility, I can not provide any information regarding the next release. It seems that @Sebi94nbg merged the pr into the development branch which seems to introduce a test matrix for the supported PHP versions. Maybe @Sebi94nbg can bring more light to that topic ^^.

**Comment by Sebi94nbg** [comment]
This change is included in the release [4.3.0](https://github.com/barbushin/php-imap/releases/tag/4.3.0) and newer.
